naresh/src/main/java/com/lftechnology/phpjava/ems/dao/EmployeeDaoImpl.java

```diff
@@ -0,0 +1,236 @@
+package com.lftechnology.phpjava.ems.dao;
+
+import com.lftechnology.phpjava.ems.entities.Employee;
+import com.lftechnology.phpjava.ems.enums.Role;
+import com.lftechnology.phpjava.ems.utlis.DbFactory;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * EmployeeDaoImpl
+ *
+ * @author Naresh Maharjan <nareshmaharjan@lftechnology.com>
+ * @since August, 08 2016
+ */
+public class EmployeeDaoImpl implements DaoSignature<Employee> {
+
+    protected Connection conn = DbFactory.getConnection();
+    protected PreparedStatement stmt = null;
+
+    /**
+     * @author Naresh Maharjan <nareshmaharjan@lftechnology.com>
+     * @return
+     * @throws SQLException
+     */
+    @Override
+    public List<Employee> findAll() throws SQLException {
+        List<Employee> employees = new ArrayList<>();
+        String sql = "SELECT * FROM employee e INNER JOIN user u ON e.user_id = u.id WHERE u.is_terminated = 0";
+        stmt = conn.prepareStatement(sql);
+        ResultSet result = stmt.executeQuery();
+        if (result.next()) {
+            Employee employee = new Employee();
```

Comment:
take out this employee definition from the loop. It will create lot of objects in case of say more than thousand users.